v2/tests/log_context_property_type_ascii_char_ptr_ut/log_context_property_type_ascii_char_ptr_ut.c

```diff
@@ -0,0 +1,530 @@
+    POOR_MANS_ASSERT(actual_call_count == expected_call_count);
+}
+
+/* Tests_SRS_LOG_CONTEXT_PROPERTY_TYPE_ASCII_CHAR_PTR_01_016: [ Otherwise, on success, `LOG_CONTEXT_PROPERTY_TYPE_GET_INIT_DATA_SIZE(ascii_char_ptr)` shall return the amount of memory needed to store the `printf` style formatted string given by `format` and the arguments in `...`. ]*/
+static void ascii_char_ptr_LOG_CONTEXT_PROPERTY_TYPE_GET_INIT_DATA_SIZE_succeeds(void)
+{
+    // arrange
+    setup_mocks();
+    setup_expected_vsnprintf_call();
+
+    // act
+    int result = LOG_CONTEXT_PROPERTY_TYPE_GET_INIT_DATA_SIZE(ascii_char_ptr)("cucu");
+
+    // assert
+    POOR_MANS_ASSERT(result == 5);
+    POOR_MANS_ASSERT(actual_and_expected_match);
+    POOR_MANS_ASSERT(actual_call_count == expected_call_count);
+}
+
+/* Tests_SRS_LOG_CONTEXT_PROPERTY_TYPE_ASCII_CHAR_PTR_01_016: [ Otherwise, on success, `LOG_CONTEXT_PROPERTY_TYPE_GET_INIT_DATA_SIZE(ascii_char_ptr)` shall return the amount of memory needed to store the `printf` style formatted string given by `format` and the arguments in `...`. ]*/
+static void ascii_char_ptr_LOG_CONTEXT_PROPERTY_TYPE_GET_INIT_DATA_SIZE_with_multiple_args_succeeds(void)
+{
+    // arrange
+    setup_mocks();
+    setup_expected_vsnprintf_call();
+
+    // act
+    int result = LOG_CONTEXT_PROPERTY_TYPE_GET_INIT_DATA_SIZE(ascii_char_ptr)("The answer is %d and let's say hello %s", 42, "world");
+
+    // assert
+    POOR_MANS_ASSERT(result == 43);
+    POOR_MANS_ASSERT(actual_and_expected_match);
+    POOR_MANS_ASSERT(actual_call_count == expected_call_count);
+}
+
+/* very "poor man's" way of testing, as no test harness and mocking framework are available */
+int main(void)
+{
+    ascii_char_ptr_to_string_with_NULL_property_value_fails();
+    ascii_char_ptr_to_string_with_NULL_buffer_and_non_zero_buffer_length_fails();
+    ascii_char_ptr_to_string_with_NULL_buffer_and_zero_buffer_length_returns_string_length();
+    ascii_char_ptr_to_string_copies_the_string();
+    when_snprintf_fails_ascii_char_ptr_to_string_also_fails();
+
+    ascii_char_ptr_copy_with_NULL_src_value_fails();
+    ascii_char_ptr_copy_with_NULL_dst_value_fails();
+    ascii_char_ptr_copy_succeeds();
+
+    ascii_char_ptr_free_returns();
+
+    ascii_char_ptr_get_type_returns_LOG_CONTEXT_PROPERTY_TYPE_ascii_char_ptr();
+
+    ascii_char_ptr_LOG_CONTEXT_PROPERTY_TYPE_INIT_with_NULL_dst_value_fails();
+    ascii_char_ptr_LOG_CONTEXT_PROPERTY_TYPE_INIT_succeeds();
+    ascii_char_ptr_LOG_CONTEXT_PROPERTY_TYPE_INIT_with_multiple_args_succeeds();
+    when_underlying_call_fails_ascii_char_ptr_LOG_CONTEXT_PROPERTY_TYPE_INIT_also_fails();
+
+    when_underlying_call_fails_ascii_char_ptr_LOG_CONTEXT_PROPERTY_TYPE_GET_INIT_DATA_SIZE_also_fails();
+    ascii_char_ptr_LOG_CONTEXT_PROPERTY_TYPE_GET_INIT_DATA_SIZE_succeeds();
+    ascii_char_ptr_LOG_CONTEXT_PROPERTY_TYPE_GET_INIT_DATA_SIZE_with_multiple_args_succeeds();
+
+    return asserts_failed;
+}
```

Comment:
consider adding "int" tests (at least for ascii_char_ptr, if not for all data types) - should exercise the code in the absence of mocks.

Reply:
Added